### ISSUE TITLE
Strange Results When Running Inference on Test Set

### ISSUE THREAD

**Issue by ItamarSafriel** [learnables/learn2learn#239]
Hello,
I tried training a classifier on a custom dataset using the ANIL algorithm. 
I used the following code in order to load my image dataset: 
```python 
data_transforms = {
    'train': transforms.Compose([
        transforms.Resize((32, 32)),
        transforms.ToTensor(),
    ]),
    'test': transforms.Compose([
        transforms.Resize((32, 32)),
        transforms.ToTensor(),
    ]),
    'val': transforms.Compose([
        transforms.Resize((32, 32)),
        transforms.ToTensor(),
    ])
}

image_datasets = {x: datasets.ImageFolder(os.path.join(data_dir, x), data_transforms[x]) for x in ["train", "test", "val"]}
class_names = image_datasets["train"].classes
dataset_dict = {x: l2l.data.MetaDataset(image_datasets[x]) for x in ["train", "test", "val"]}

l2l_transforms = {
    "train": [FusedNWaysKShots(dataset_dict["train"], n=n_ways, k=2*k_shots),
              LoadData(dataset_dict["train"]),
              RemapLabels(dataset_dict["train"]),
              ConsecutiveLabels(dataset_dict["train"])],
            
    "test": [FusedNWaysKShots(dataset_dict["test"], n=n_ways, k=2*k_shots),
            LoadData(dataset_dict["test"]),
            RemapLabels(dataset_dict["test"]),
            ConsecutiveLabels(dataset_dict["test"])],

    "val": [FusedNWaysKShots(dataset_dict["val"], n=n_ways, k=2*k_shots),
            LoadData(dataset_dict["val"]),
            RemapLabels(dataset_dict["val"]),
            ConsecutiveLabels(dataset_dict["val"])]
}

tasksets = {x: l2l.data.TaskDataset(dataset_dict[x], l2l_transforms[x], num_tasks) for x in ["train", "test", "val"]}
```
And trained using the same code as in the anil_fc100.py example (with added saving of the learner state dict). 
I also logged the train,test and val accuracy and error and ended up with these graphs (for a very short training of 10 iterations): 
![image](https://user-images.githubusercontent.com/49122015/116098115-78254f80-a6b3-11eb-8abb-dfb0032f3a4b.png)
As you can see, the train and test accuracy is very high (almost a 100%) and the validation accuracy is decent (around 75%)

I tried to inference on the test set using the following code: 
```python 
  for i in range(0, 20):
    batch = tasksets["test"].sample()
    inputs, labels = batch
    labels = labels.to(device)
    inputs = inputs.to(device)
    extracted_features = features(inputs)
    class_probs = head(extracted_features).detach().to("cpu").numpy()
    pred_idx = np.argmax(class_probs, axis=1)
    labels = labels.detach().to("cpu").numpy()
    acc = (pred_idx == labels).tolist().count(True)/len(labels)*100
    print("acc: ", acc)
``` 
and got the following accuracies (for 20 runs): 100, 100, 0, 100, 100, 100, 0, 0, 0, 0, 100, 0, 100, 100, 100, 0, 100, 0, 100, 0 (%) while the pred_idx returns the same array every time [1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] 

This is very strange, does any one have any idea what went wrong? 
The dataset has two classes with 1000 images each.  

Thanks in advance!

**Comment by seba-1511** [comment]
Hello @ItamarSafriel,

It seems like you are not adapting the head for classification, try: `head.adapt(loss)` before computing the final accuracies, where `loss` is the cross-entropy on the adaptation (query) set.

**Comment by ItamarSafriel** [comment]
Hello @seba-1511 and thank you very much for your response! 

There already was an adaptation of the head in my code, but I think I found the problem. The transform of ```RemapLabels``` has shuffled the labels for some reason, and when I deleted it from 
l2l_transforms everything worked as expected. Is this behaviour of ```RemapLabels``` is on purpose? If so, what is the correct way to use it?

Thanks in advance!
